### PR TITLE
Make Cosmos DB, SQL Pool  & Stream Analytics Deployment Optional

### PR DESCRIPTION
**Pull request by marvinbuss**
## Summary of the Pull Request
This PR will make the deployment of Cosmos DB, Synapse SQL Pool and Stream Analytics optional. Users can select whether these should be enabled as part of the deployment.

## References
None

## PR Checklist
* [x] Closes Issue https://github.com/Azure/data-product-batch/issues/98
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA

## Validation Steps Performed
See Link and images in comments below.

### CODE REVIEW

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 1]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
 | tags | Specifies the tags that you want to apply to all resources. | {`key`: `value`} |
 | administratorPassword | Specifies the administrator password of the sql servers. Will be automatically set in the workflow. **Leave this value as is.** | `<your-secure-password>` |
 | synapseDefaultStorageAccountFileSystemId | Specifies the resource ID of the default storage account file system for synapse. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Storage/storageAccounts/{storage-name}/blobServices/default/containers/{container-name}` |
+| enableSqlPool | Specifies whether an Azure SQL Pool should be deployed inside your Synapse workspace as part of the template. If you selected dataFactory as processingService, leave this value as is. | `true` or `false` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:80]: Minor updates add backticks please.

`enableSqlPool`
`processingService`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 2]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
+| enableSqlServer | Specifies whether Azure SQL Server should be deployed as part of the template. | `true` or `false` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:81]: Add backticks for `enableSqlServer`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 3]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
+| enableCosmos | Specifies whether Azure Cosmos DB should be deployed as part of the template. | `true` or `false` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:82]: `enableCosmos`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 4]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
+| enableStreamAnalytics | Specifies whether Azure Stream Analytics Cluster and Job should be deployed as part of the template. | `true` or `false` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:83]: `enableStreamAnalytics`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 5]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
 | streamanalyticsDefaultStorageAccountFileSystemId | Specifies the resource ID of the default storage account  file system for stream analytics. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Storage/storageAccounts/{storage-name}/blobServices/default/containers/{container-name}` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:84]: `streamanalyticsDefaultStorageAccountFileSystemId`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 6]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
 | subnetId | Specifies the resource ID of the subnet to which all services will connect. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Network/virtualNetworks/{vnet-name}/subnets/{subnet-name}` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:85]: `subnetId`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 7]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
 | purviewId | Specifies the resource ID of the central purview instance. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Purview/accounts/{purview-name}` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:86]: `purviewId`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 8]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
-| enableRoleAssignments | Specifies whether role assignments should be enabled. **Leave this value as is.** | `true` or `false` |
+| enableRoleAssignments | Specifies whether role assignments should be enabled. | `true` or `false` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:87]: `enableRoleAssignments`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 9]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
 | privateDnsZoneIdKeyVault | Specifies the resource ID of the private DNS zone for KeyVault. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Network/privateDnsZones/privatelink.vaultcore.azure.net` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:88]: `privateDnsZoneIdKeyVault`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 10]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
 | privateDnsZoneIdSynapseDev | Specifies the resource ID of the private DNS zone for Synapse Dev. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Network/privateDnsZones/privatelink.dev.azuresynapse.net` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:89]: `privateDnsZoneIdSynapseDev`

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md [hunk 11]
```diff
@@ -77,10 +77,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
...
 | privateDnsZoneIdSynapseSql | Specifies the resource ID of the private DNS zone for Synapse Sql. | `/subscriptions/{subscription-id}/resourceGroups/{rg-name}/providers/Microsoft.Network/privateDnsZones/privatelink.sql.azuresynapse.net` |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:90]: `privateDnsZoneIdSynapseSql`

File: docs/EnterpriseScaleAnalytics-GitHubActionsDeployment.md [hunk 12]
```diff
@@ -68,10 +68,14 @@ To begin, please open the [infra/params.dev.json](/infra/params.dev.json). In th
 | tags | Specifies the tags that you want to apply to all resources. | {`key`: `value`} |
```
Comment (amanjeetsingh) [inline docs/EnterpriseScaleAnalytics-GitHubActionsDeployment.md:68]: `tags`

**Review by amanjeetsingh — COMMENTED** [review]
Some minor changes.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 2 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:81]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 4 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:83]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 6 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:85]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 8 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:87]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 10 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:89]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-GitHubActionsDeployment.md (hunk 12 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-GitHubActionsDeployment.md:68]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 11 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:90]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 9 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:88]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 7 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:86]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 5 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:84]: Let's get this done as part of another PR.

File: docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md (hunk 3 above)
Comment (marvinbuss) [reply docs/EnterpriseScaleAnalytics-AzureDevOpsDeployment.md:82]: Let's get this done as part of another PR.

**Review by marvinbuss — COMMENTED** [review]
LGTM!

**Review by amanjeetsingh — APPROVED** [review]

**Review by amanjeetsingh — APPROVED** [review]

**Pull request merged**